rest-servlet/src/test/java/org/hawkular/inventory/rest/filters/ParamPropertiesTest.java

```diff
@@ -0,0 +1,97 @@
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import org.hawkular.inventory.rest.filters.ParamProperties.Parser;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class ParamPropertiesTest {
+    @Test
+    public void testParser() {
+        /* escaping */
+        assertParse("\\:k1:v1,k2:v2", ":k1", "v1", "k2", "v2");
+        assertParse("k\\:1:v1,k2:v2", "k:1", "v1", "k2", "v2");
+        assertParse("k1\\::v1,k2:v2", "k1:", "v1", "k2", "v2");
+        assertParse("k1:\\:v1,k2:v2", "k1", ":v1", "k2", "v2");
+        assertParse("k1:v\\:1,k2:v2", "k1", "v:1", "k2", "v2");
+        assertParse("k1:v1\\:,k2:v2", "k1", "v1:", "k2", "v2");
+        assertParse("k1:v1,\\:k2:v2", "k1", "v1", ":k2", "v2");
+
+        assertParse("\\,k1:v1,k2:v2", ",k1", "v1", "k2", "v2");
+        assertParse("k\\,1:v1,k2:v2", "k,1", "v1", "k2", "v2");
+        assertParse("k1\\,:v1,k2:v2", "k1,", "v1", "k2", "v2");
+        assertParse("k1:\\,v1,k2:v2", "k1", ",v1", "k2", "v2");
+        assertParse("k1:v\\,1,k2:v2", "k1", "v,1", "k2", "v2");
+        assertParse("k1:v1\\,,k2:v2", "k1", "v1,", "k2", "v2");
+        assertParse("k1:v1,\\,k2:v2", "k1", "v1", ",k2", "v2");
+
+        assertParse("\\\\k1:v1,k2:v2", "\\k1", "v1", "k2", "v2");
+        assertParse("k\\\\1:v1,k2:v2", "k\\1", "v1", "k2", "v2");
+        assertParse("k1\\\\:v1,k2:v2", "k1\\", "v1", "k2", "v2");
+        assertParse("k1:\\\\v1,k2:v2", "k1", "\\v1", "k2", "v2");
+        assertParse("k1:v\\\\1,k2:v2", "k1", "v\\1", "k2", "v2");
+        assertParse("k1:v1\\\\,k2:v2", "k1", "v1\\", "k2", "v2");
+        assertParse("k1:v1,\\\\k2:v2", "k1", "v1", "\\k2", "v2");
+
+        /* mainstream cases */
+        assertParse("k1:v1", "k1", "v1");
+        assertParse("k1:v1,k2:v2", "k1", "v1", "k2", "v2");
+
+        /* multivalue */
+        assertParse("k1:v1,k1:v2", "k1", new String[] {"v1", "v2"});
+
+        /* with various parts empty */
+        assertParse("");
+        assertParse("k1,k2:v2", "k1", null, "k2", "v2");
+        assertParse("k1:,k2:v2", "k1", "", "k2", "v2");
+        assertParse(":v1,k2:v2", "", "v1", "k2", "v2");
+        assertParse(":,k2:v2", "", "", "k2", "v2");
+
+        /* spaces - no trimming */
+        assertParse("k1: v1, k2:v2", "k1", " v1", " k2", "v2");
+        assertParse("k 1:v1,k2:v 2", "k 1", "v1", "k2", "v 2");
+
+
+    }
+    private void assertParse(String serialized, Object... keys) {
+        LinkedHashMap<String, List<String>> expected = new LinkedHashMap<>();
+        for (int i = 0; i < keys.length; ) {
+            String key = (String) keys[i++];
+            Object v = keys[i++];
+            if (v == null) {
+                expected.put(key, null);
+            } else {
+                String[] valArray = v instanceof String ? new String[] {(String) v} : (String[]) v;
+                List<String> valList = new ArrayList<>();
```

Comment:
Actually +1 here in this test. Otherwise, I tend to avoid `Arrays.asList()` because it returns a writable `java.util.Arrays.ArrayList` instance backed by the array. Because the array can leak, one cannot reliably make the ArrayList immutable. But you are right that it is not a problem here.